api/envoy/config/route/v3/route_components.proto

```diff
@@ -2391,6 +2392,14 @@ message InternalRedirectPolicy {
   // Allow internal redirect to follow a target URI with a different scheme than the value of
   // x-forwarded-proto. The default is false.
   bool allow_cross_scheme_redirect = 4;
+
+  // Specifies a list of headers, by name, to copy from the internal redirect into the subsequent
+  // request. If a header is specified here but not present in the redirect, it will be cleared in
```

Comment:
Usually for headers we explicitly disallow special HTTP/2 headers (:host :scheme etc).  Again question for API shepherds if we want to be consistent here?

Reply:
I think it probably depends on the use case. If we want to block that in code yeah we should document here.

Reply:
There's not much point in adding those into this list, since they're already copied.... but there's also no sense in letting users do silly and confusing things.

Is there a standard pattern for disallowing these that I should follow?

Reply:
for config validation we use HeaderUtility::isModifiableHeader and reject those if configured.  not seeing useful comments to crib off of.

Reply:
added a check using isModifiableHeader in config_impl.cc; I'm reasonably sure that's the right place.